File: backend/src/main/java/com/yourorg/appname/service/AppointmentService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.AppointmentRequest;
import com.yourorg.appname.dto.request.RescheduleRequest;
import com.yourorg.appname.dto.response.AppointmentResponse;

import java.util.List;

public interface AppointmentService {
    AppointmentResponse createAppointment(AppointmentRequest request, String currentUsername);
    AppointmentResponse getAppointmentById(Long id);
    AppointmentResponse getAppointmentByReference(String reference);
    List<AppointmentResponse> getAppointmentsForCurrentPatient(String currentUsername);
    AppointmentResponse rescheduleAppointment(Long id, RescheduleRequest request, String currentUsername);
    AppointmentResponse cancelAppointment(Long id, String currentUsername);
    List<AppointmentResponse> getAllAppointments();
}

File: frontend/src/services/appointmentService.js
import apiClient from './apiClient';

export const appointmentService = {
  async createAppointment(appointmentData) {
    const response = await apiClient.post('/appointments', appointmentData);
    return response.data;
  },

  async getMyAppointments() {
    const response = await apiClient.get('/appointments/my');
    return response.data;
  },

  async getAppointmentById(id) {
    const response = await apiClient.get(`/appointments/${id}`);
    return response.data;
  },

  async getAppointmentByReference(reference) {
    const response = await apiClient.get(`/appointments/reference/${reference}`);
    return response.data;
  },

  async rescheduleAppointment(id, newDate, newTimeSlot) {
    const response = await apiClient.put(`/appointments/${id}/reschedule`, {
      newDate,
      newTimeSlot,
    });
    return response.data;
  },

  async cancelAppointment(id) {
    const response = await apiClient.put(`/appointments/${id}/cancel`);
    return response.data;
  },
};
